src/components/adslot-ui/ToggleSwitch/index.jsx

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import './style.scss';
+
+class ToggleSwitch extends React.Component {
+  state = {
+    value: _.get(this.props, 'defaultValue', false),
+  };
+
+  handleChange = event => {
+    const eventValue = _.get(event, 'target.checked');
+    const { onChange } = this.props;
+
+    if (onChange) {
```

Comment:
You are checking if `onChange` exists or not. You should check  `_.isNil(this.props.value)` which will then throw a console error on if `onChange` does not exists. Which will in turn serve as a guard on top of warning. 

```suggestion
    if (_.isNil(this.props.value)) {
      this.setState({ value: eventValue });
    } else {
      onChange(eventValue);
    }
```